src/goto-programs/goto_convert_side_effect.cpp

```diff
@@ -506,12 +511,13 @@ void goto_convertt::remove_malloc(
     call.move_to_operands(expr);
   }
 
-  convert(call, dest);
+  convert(call, dest, mode);
 }
 
 void goto_convertt::remove_temporary_object(
   side_effect_exprt &expr,
   goto_programt &dest,
+  const irep_idt &mode,
```

Comment:
This one isn't necessary, `temporary_object` actually does have a mode provided.